src/ast.rs

```diff
@@ -72,50 +73,47 @@ impl From<((usize, usize, usize), (usize, usize, usize))> for Span {
 }
 
 /// Represents a name of an entity, such as a type, variable, function, etc.
-#[derive(Debug, PartialEq)]
-pub struct RawIdentifier<'a> {
-    pub name: &'a str,
-}
+#[derive(Debug, PartialEq, Clone)]
+pub struct RawIdentifier(pub String);
 
-pub type Identifier<'a> = Spanned<RawIdentifier<'a>>;
+pub type Identifier = Spanned<RawIdentifier>;
 
-impl<'a> From<ParseInput<'a>> for RawIdentifier<'a> {
+impl<'a> From<ParseInput<'a>> for RawIdentifier {
     fn from(span: ParseInput<'a>) -> Self {
-        RawIdentifier {
-            name: span.fragment(),
-        }
+        RawIdentifier((*span.fragment()).to_owned())
     }
 }
 
-/// A parameter to a function, e.g., `foo: MyType`.
-#[derive(Debug, PartialEq)]
-pub struct RawParameter<'a> {
-    pub name: Identifier<'a>,
-    pub param_type: Identifier<'a>,
+/// Introduces a variable with its name and type, e.g., `foo : MyType`. E.g. used in function
```

Comment:
Nit: this is one spot where DSLX deviates, probably for the worse, from Rust, in that values are by default mutable. In fact, one could hold the opinion that all values/names should _always and only_ be immutable in DSLX. The upshot being: "variable" might be a term we use with care, and perhaps just avoid if possible.

If we stay this route, maybe add a TODO to this effect?

Reply:
I think we have to match C++ DSLX's existing semantics, more or less exactly, right?

> one could hold the opinion that all values/names should always and only be immutable in DSLX

That would be my strong opinion, but my opinion is irrelevant in light of the above (if above is true).

> If we stay this route, maybe add a TODO to this effect?

If we have to match C++, then this should remain named "variable", right? 

If not, what name? `BindingDeclaration`?

Reply:
1) Yeah. TODO or create an issue to consider "fixing" this once we're at parity? IDK
2) :)
3) I'd prefer something like that, but as usual, I'm happy to TODO it or whatever. After all, we have bulk code modification tools like ~Rosie~ sed.

Reply:
1. Opened https://github.com/xls-friends/dslx-rust/issues/29
3. Renamed to `RawBindingDecl`, related types also renamed.